tests/testthat/test-dofv.R

```diff
@@ -0,0 +1,232 @@
+  tdir <- local_model_dir_nonmem_dummies(
+    list(
+      list(
+        subpath = "01",
+        data_set = "../data.csv",
+        final_est_method = "First Order Conditional Estimation with Interaction",
+        number_of_obs = 800,
+        ofv_no_constant = ofvs[1]
+      ),
+      list(
+        subpath = "02",
+        based_on = "01",
+        data_set = "../data.csv",
+        final_est_method = "First Order Conditional Estimation with Interaction",
+        number_of_obs = 800,
+        ofv_no_constant = ofvs[2]
+      ),
+      list(
+        # First based_on item is used as reference.
+        subpath = "03",
+        based_on = c("01", "02"),
+        data_set = "../data.csv",
+        final_est_method = "First Order Conditional Estimation with Interaction",
+        number_of_obs = 800,
+        ofv_no_constant = ofvs[3]
+      ),
+      list(
+        # Second-level directory works.
+        subpath = "sub/04",
+        based_on = "../03",
+        data_set = "../../data.csv",
+        final_est_method = "First Order Conditional Estimation with Interaction",
+        number_of_obs = 800,
+        ofv_no_constant = ofvs[4]
+      ),
+      list(
+        # dofv is set to NA due to different number_of_obs.
+        subpath = "05",
+        based_on = "03",
+        data_set = "../data.csv",
+        final_est_method = "First Order Conditional Estimation with Interaction",
+        number_of_obs = 900,
+        ofv_no_constant = ofvs[5]
+      ),
+      list(
+        # dofv is set to NA due to different data path.
+        subpath = "06",
+        based_on = "03",
+        data_set = "../foo.csv",
+        final_est_method = "First Order Conditional Estimation with Interaction",
+        number_of_obs = 800,
+        ofv_no_constant = ofvs[6]
+      ),
+      list(
+        # dofv is set to NA due to different final method.
+        subpath = "07",
+        based_on = "03",
+        data_set = "../data.csv",
+        final_est_method = "Laplacian Conditional Estimation (Centered)",
+        ofv_no_constant = ofvs[7]
+      ),
+      list(
+        # dofv is non-NA for matching methods.
+        subpath = "08",
+        based_on = "07",
+        data_set = "../data.csv",
+        final_est_method = "Laplacian Conditional Estimation (Centered)",
+        ofv_no_constant = ofvs[8]
+      ),
+      list(
+        # dofv is NA due unsupported method, plus the method doesn't match the
+        # base's.
+        subpath = "09",
+        based_on = "03",
+        data_set = "../data.csv",
+        final_est_method = "Importance Sampling of Variance-Covariance (SIR)",
+        ofv_no_constant = ofvs[9]
+      ),
+      list(
+        # dofv is set to NA due to unsupported method, despite base's matching
+        # method.
+        subpath = "10",
+        based_on = "09",
+        data_set = "../data.csv",
+        final_est_method = "Importance Sampling of Variance-Covariance (SIR)",
+        ofv_no_constant = ofvs[10]
```

Comment:
Comprehensive test suite - checks all the cases I would think to
 - tests confirm we support `run_log` and `run_log() %>% add_summary()`. Given the handling *I don't* think we necessarily need to test `summary_log() %>% add_dofv()`, though just noting that we don't currently (I tested manually, though given that we use `summary_log_impl` under the hood it makes sense that we don't really need to test that call too)